### PR TITLE
Adyen: Replace empty state string with N/A

### PR DESCRIPTION
**Pull request by therufs**
[ECS-367](https://spreedly.atlassian.net/browse/ECS-367)

### CODE REVIEW

File: test/remote/gateways/remote_adyen_test.rb [hunk 1]
```diff
@@ -605,6 +605,13 @@ def test_blank_country_for_purchase
     assert_match Gateway::STANDARD_ERROR_CODE[:incorrect_address], response.error_code
   end
 
+  def test_nil_state_for_purchase
+    @options[:billing_address][:state] = nil
+    response = @gateway.authorize(@amount, @credit_card, @options)
+    assert_failure response
+    assert_match Gateway::STANDARD_ERROR_CODE[:incorrect_address], response.error_code
+  end
+
```
Comment (molbrown) [inline test/remote/gateways/remote_adyen_test.rb]: 🤔 Wait a minute. Don't we want both these state tests to succeed? Are we sure N/A is what Adyen wants here?

File: lib/active_merchant/billing/gateways/adyen.rb [hunk 2]
```diff
@@ -231,11 +231,15 @@ def add_address(post, options)
...
-          post[:billingAddress][:stateOrProvince] = address[:state] || 'N/A'
+          post[:billingAddress][:stateOrProvince] = get_state(address)
           post[:billingAddress][:country] = address[:country] if address[:country]
         end
       end
 
+      def get_state(address)
+        address[:state] && !address[:state].blank? ? address[:state] : 'N/A'
+      end
+
```
Comment (molbrown) [inline lib/active_merchant/billing/gateways/adyen.rb]: 😻

File: test/remote/gateways/remote_adyen_test.rb (hunk 1 above)
Comment (therufs) [reply test/remote/gateways/remote_adyen_test.rb]: *facepalm* yes I suppose we do want that

**Comment by therufs** [comment]
Unit tests:
--------------------------------------------------------------------------------------------------------------------------------
38 tests, 183 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
--------------------------------------------------------------------------------------------------------------------------------

Remote tests:
--------------------------------------------------------------------------------------------------------------------------------
61 tests, 187 assertions, 4 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
93.4426% passed
--------------------------------------------------------------------------------------------------------------------------------

Fixes for failing tests anticipated pursuant to https://spreedly.slack.com/archives/CF080246R/p1561051065014400

File: test/remote/gateways/remote_adyen_test.rb (hunk 1 above)
Comment (molbrown) [reply test/remote/gateways/remote_adyen_test.rb]: I'm curious. How does this just work now? Did you have to change an account setting or something?

Comment (therufs) [reply test/remote/gateways/remote_adyen_test.rb]: Hmm, the test has been flipped to expect success but the update doesn't seem to be self-aware in Github 🤔

**Pull request merged**